processing/src/main/java/io/druid/query/groupby/epinephelinae/RowBasedGrouperHelper.java

```diff
@@ -172,10 +182,27 @@
           if (isInputRaw) {
             IndexedInts index = dimensionSelectors[i - dimStart].getRow();
             value = index.size() == 0 ? "" : dimensionSelectors[i - dimStart].lookupName(index.get(0));
+            key[i] = Strings.nullToEmpty(value);
           } else {
-            value = (String) row.getRaw(query.getDimensions().get(i - dimStart).getOutputName());
+            // only V2 groupby supports non-String types
```

Comment:
my intent was to clarify the condition there, since I thought it wasn't immediately apparent that the `isInputRaw == true` case applied to groupby V2